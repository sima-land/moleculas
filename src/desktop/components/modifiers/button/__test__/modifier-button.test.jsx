import React from 'react';
import { mount, shallow } from 'enzyme';
import { ModifierButton } from '..';

describe('<ModifierButton />', () => {
  it('should render without props', () => {
    const wrapper = mount(
      <ModifierButton />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render count', () => {
    const wrapper = mount(
      <ModifierButton
        count={12}
      />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ count: undefined });

    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ count: 123 });

    expect(wrapper).toMatchSnapshot();
  });

  it('should render markdown and count', () => {
    const wrapper = mount(
      <ModifierButton
        count={666}
        isMarkdown
      />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ count: undefined });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle "selected" properly', () => {
    const wrapper = mount(
      <ModifierButton
        selected
      />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ selected: undefined });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle "color" properly', () => {
    const wrapper = mount(
      <ModifierButton
        color='#bada55'
      />
    );
    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ color: undefined });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle "type" and "content" properly', () => {
    const wrapper = mount(
      <ModifierButton
        type='text'
        content='Test content'
      />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ type: 'image', image: 'www.abc.com' });
    expect(wrapper).toMatchSnapshot();

    wrapper.setProps({ type: undefined, content: undefined });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle "className" properly', () => {
    const wrapper = mount(
      <ModifierButton
        className='test-class'
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should not show title when type is color', () => {
    const wrapper = mount(
      <ModifierButton
        type='color'
        content='content'
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle "onClick" prop', () => {
    const spy = jest.fn();

    const wrapper = shallow(
      <ModifierButton
        className='test-class'
        content='Test content'
        type='text'
        onClick={spy}
      />
    );

    expect(wrapper).toMatchSnapshot();
    expect(spy).toBeCalledTimes(0);

    wrapper.find('.modifier-button').simulate('click');
    expect(spy).toBeCalledTimes(1);
  });
});
