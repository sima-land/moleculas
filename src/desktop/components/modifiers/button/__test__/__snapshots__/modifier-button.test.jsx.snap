// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModifierButton /> should handle "className" properly 1`] = `
<ForwardRef(ModifierButton)
  className="test-class"
>
  <div
    className="modifier-button test-class"
  >
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should handle "color" properly 1`] = `
<ForwardRef(ModifierButton)
  color="#bada55"
>
  <div
    className="modifier-button"
    style={
      Object {
        "background": "#bada55",
      }
    }
  >
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should handle "color" properly 2`] = `
<ForwardRef(ModifierButton)>
  <div
    className="modifier-button"
  >
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should handle "onClick" prop 1`] = `
<div
  className="modifier-button test-class"
  onClick={[MockFunction]}
>
  <div
    className="content"
  >
    <span
      className="text guttered"
    >
      Test content
    </span>
  </div>
</div>
`;

exports[`<ModifierButton /> should handle "selected" properly 1`] = `
<ForwardRef(ModifierButton)
  selected={true}
>
  <div
    className="modifier-button selected"
  >
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should handle "selected" properly 2`] = `
<ForwardRef(ModifierButton)>
  <div
    className="modifier-button"
  >
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should handle "type" and "content" properly 1`] = `
<ForwardRef(ModifierButton)
  content="Test content"
  type="text"
>
  <div
    className="modifier-button"
  >
    <div
      className="content"
    >
      <span
        className="text guttered"
      >
        Test content
      </span>
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should handle "type" and "content" properly 2`] = `
<ForwardRef(ModifierButton)
  content="Test content"
  image="www.abc.com"
  type="image"
>
  <div
    className="modifier-button squared"
  >
    <div
      className="content"
    >
      <img
        alt="Test content"
        className="image"
        src="www.abc.com"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should handle "type" and "content" properly 3`] = `
<ForwardRef(ModifierButton)
  image="www.abc.com"
>
  <div
    className="modifier-button"
  >
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should not show title when type is color 1`] = `
<ForwardRef(ModifierButton)
  content="content"
  type="color"
>
  <div
    className="modifier-button squared"
  />
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should render count 1`] = `
<ForwardRef(ModifierButton)
  count={12}
>
  <div
    className="modifier-button"
  >
    <span
      className="label with-count"
    >
      12
    </span>
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should render count 2`] = `
<ForwardRef(ModifierButton)>
  <div
    className="modifier-button"
  >
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should render count 3`] = `
<ForwardRef(ModifierButton)
  count={123}
>
  <div
    className="modifier-button"
  >
    <span
      className="label with-count"
    >
      99+
    </span>
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should render markdown and count 1`] = `
<ForwardRef(ModifierButton)
  count={666}
  isMarkdown={true}
>
  <div
    className="modifier-button"
  >
    <span
      className="label with-count"
    >
      99+
      (У)
    </span>
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should render markdown and count 2`] = `
<ForwardRef(ModifierButton)
  isMarkdown={true}
>
  <div
    className="modifier-button"
  >
    <span
      className="label"
    >
      У
    </span>
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;

exports[`<ModifierButton /> should render without props 1`] = `
<ForwardRef(ModifierButton)>
  <div
    className="modifier-button"
  >
    <div
      className="content"
    >
      <span
        className="text guttered"
      />
    </div>
  </div>
</ForwardRef(ModifierButton)>
`;
